### PR TITLE
Migrate MSBuild plugin to CoreCLR based Microsoft.Build packages.

### PR DESCRIPTION
**Pull request by troydai**

### CODE REVIEW

**Comment by troydai** [comment]
I'm cleaning up the changes now.

**Comment by troydai** [comment]
@hal-ler can you please take a look why the mac CI build keep failing?

**Comment by hal-ler** [comment]
The NuGet version (used for Cake) was hard-coded to 3.3.0 because the newer version times out on Mac.

**Comment by troydai** [comment]
ah, let me revert it back to 3.3.0

File: tests/OmniSharp.Roslyn.CSharp.Tests/GoToDefinitionFacts.cs [hunk 1]
```diff
@@ -99,7 +99,7 @@ class Foo {
 
             Assert.Null(definitionResponse.FileName);
             Assert.NotNull(definitionResponse.MetadataSource);
-            Assert.Equal("mscorlib", definitionResponse.MetadataSource.AssemblyName);
+            Assert.Equal("System.Private.CoreLib", definitionResponse.MetadataSource.AssemblyName);
```
Comment (david-driscoll) [inline tests/OmniSharp.Roslyn.CSharp.Tests/GoToDefinitionFacts.cs]: Does this work for net451 based projects as well?

Comment (troydai) [inline tests/OmniSharp.Roslyn.CSharp.Tests/GoToDefinitionFacts.cs]: No. But we're not running the test on full desktop now.

**Comment by hal-ler** [comment]
The new error is something I never saw before. Is there an issue with the Mac CLI package? It seems to be missing the Crypto DLL from the error message.

**Comment by troydai** [comment]
Related to https://github.com/dotnet/corefx/issues/9171

**Comment by david-driscoll** [comment]
Once it passes ci

LGTM

**Comment by DustinCampbell** [comment]
oh wow! @troydai == rock god!

**Pull request merged**